### PR TITLE
fix: Move sample repo to Amplify UI branch

### PR DESCRIPTION
**Pull request by ErikCH**
#### Description of changes
Added a new sample guides app to our repo with an e2e test. I updated the link in our [Protected Route Guide](https://ui.docs.amplify.aws/react/guides/auth-protected) so it points to the new folder in our repo. 

In a future PR I'll add in the workflow file to automatically run the test on push to main.

#### Issue #, if available
#1925 

#### Description of how you validated changes
Created e2e test

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/guides/auth-protected/auth-protected.react.mdx [hunk 1]
```diff
@@ -9,7 +9,7 @@ Let's imagine you needed a web application where the login page was in its own r
 
 This guide demonstrates that scenario while using the [Amplify UI Authenticator](/components/authenticator) and the [Amplify CLI](https://docs.amplify.aws/cli/).
 
-> If you'd like to follow along, please use this example [Github repo](https://github.com/ErikCH/protected-route-amplify-react). Additionally, you can watch this [video](https://youtu.be/6f635LVtwgk) that walks through this process, and includes a bonus section on how to create a lambda function with a custom authorizer.
+> If you'd like to follow along, please use this example [Github repo](https://github.com/aws-amplify/amplify-ui/tree/main/guides/react/protected-routes). Additionally, you can watch this [video](https://youtu.be/6f635LVtwgk) that walks through this process, and includes a bonus section on how to create a lambda function with a custom authorizer.
```
Comment (ErikCH) [inline docs/src/pages/[platform]/guides/auth-protected/auth-protected.react.mdx:12]: Updated link here

File: guides/react/protected-routes/package.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    "react-scripts": "5.0.0",
+    "web-vitals": "^2.1.4"
+  },
+  "scripts": {
+    "dev": "react-app-rewired start",
+    "start": "serve build -s -p 3000",
+    "build": "react-app-rewired build",
+    "postbuild": "yarn run size-limit",
+    "test": "react-app-rewired test",
+    "eject": "react-app-rewired eject"
```
Comment (ErikCH) [inline guides/react/protected-routes/package.json:23]: I use this so we can import the aws-exports file outside of this cra app

**Review by slaymance — APPROVED** [review]
Pulled the branch and tested manually and everything seems good. I did notice that when running `yarn build`, I got the error `error Command "size-limit" not found.` when it ran the postbuild script. However, this didn't cause the build to fail and `yarn start` worked fine.

**Comment by ErikCH** [comment]
> Pulled the branch and tested manually and everything seems good. I did notice that when running `yarn build`, I got the error `error Command "size-limit" not found.` when it ran the postbuild script. However, this didn't cause the build to fail and `yarn start` worked fine.

Nice catch! I accidentally left in a post-build script in the package.json file. I removed it.

**Review by wlee221 — APPROVED** [review]
LGTM!

**Pull request merged**